spring-cloud-azure-sql-sample failed

Configuration properties:

```properties
spring.cloud.azure.credentialFilePath=my-auth-file
spring.cloud.azure.resourceGroup=xxx-east
spring.cloud.azure.region=eastUS

spring.cloud.azure.sql.database-name=sql-db
# The server name can only be made up of lowercase letters, the numbers and the hyphen(-).
spring.cloud.azure.sql.server-name=sql-server

# Leave username empty if database server already created
spring.datasource.username=dbusername

spring.datasource.password=dbpassword

# So app starts despite "table already exists" errors.
spring.datasource.continue-on-error=true
# Enforces database initialization
spring.datasource.initialization-mode=always
```

Failure stacktrace:
```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.autoconfigure.jdbc.DataSourceProperties]: Factory method 'cloudSqlDataSourceProperties' threw exception; nested exception is java.lang.RuntimeException: getOrCreateSqlServer with name 'sql-server' failed due to: Async operation failed with provisioning state: Failed
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:579)
	... 69 more
Caused by: java.lang.RuntimeException: getOrCreateSqlServer with name 'sql-server' failed due to: Async operation failed with provisioning state: Failed
	at com.microsoft.azure.spring.cloud.context.core.ResourceCreatingAspect.handleExceptionAndLog(ResourceCreatingAspect.java:57)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:644)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:633)

```

--- a/spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/ResourceCreatingAspect.java
+++ b/spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/ResourceCreatingAspect.java
@@ -52,9 +52,10 @@ public class ResourceCreatingAspect {
         try {
             return joinPoint.proceed();
         } catch (CloudException e) {
-            LOG.error("{} with name '{}' failed due to: {}", methodName, name, e.getMessage());
+            String errorMessage = String.join(", ", e.getMessage(), e.body().code(), e.body().message());
+            LOG.error("{} with name '{}' failed due to: {}", methodName, name, errorMessage);
             throw new RuntimeException(
-                    String.format("%s with name '%s' failed due to: %s", methodName, name, e.getMessage()));
+                    String.format("%s with name '%s' failed due to: %s", methodName, name, errorMessage));
         } finally {
             stopWatch.stop();
             LOG.info("{} with name '{}' finished in {} seconds", methodName, name, stopWatch.getTotalTimeSeconds());

--- a/spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/ResourceCreatingAspect.java
+++ b/spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/ResourceCreatingAspect.java
@@ -53,9 +53,10 @@ public class ResourceCreatingAspect {
             return joinPoint.proceed();
         } catch (CloudException e) {
             String errorMessage = String.join(", ", e.getMessage(), e.body().code(), e.body().message());
-            LOG.error("{} with name '{}' failed due to: {}", methodName, name, errorMessage);
-            throw new RuntimeException(
-                    String.format("%s with name '%s' failed due to: %s", methodName, name, errorMessage));
+            String message = String.format("%s with name '%s' failed due to: %s", methodName, name, errorMessage);
+
+            LOG.error(message);
+            throw new RuntimeException(message);
         } finally {
             stopWatch.stop();
             LOG.info("{} with name '{}' finished in {} seconds", methodName, name, stopWatch.getTotalTimeSeconds());

